Fwa 008 weather detail box

heupotato opened the pull request.

heupotato committed:
Add file mock json data

heupotato committed:
Create models

heupotato committed:
Generate models by json_model package

heupotato committed:
Update home_screen.dart

heupotato committed:
Update home_screen.dart

heupotato committed:
Remove mockjsondata

heupotato committed:
Move extension of Weather to model packages

heupotato committed:
Update weather_extension.dart

Move methods to WeekWeather extension

heupotato committed:
Update weather_extension.dart

Create class DayWeather and WeekWeather to handle data

heupotato committed:
Update weather_day_detail_widget.dart

heupotato committed:
Update weather_extension.dart

heupotato committed:
Delete mock_json_data.json

heupotato committed:
Update weather_extension.dart

heupotato committed:
Update weather_day_detail_widget.dart

heupotato committed:
Update weather_extension.dart

heupotato committed:
Update weather_extension.dart

Add typicalWeather getter

heupotato committed:
Update weather_extension.dart

- Add cloudCoverage , precipitation, windSpeed, windDirection getter (WeatherData class)
- Add today and tonight related data getter (class DayWeather)

heupotato committed:
Create and Add weather detail box to homescreen

lngyeen commented on Flutter_Weather/lib/widgets/weather_detail_box_widget.dart:
```diff
@@ -0,0 +1,130 @@
+    return Text(
+        "Tonight is ${tonightData.weather}. "
+            "Winds ${todayWeather.tonightWindDirection} at ${tonightData.windSpeed}."
+            "The overnight low will be ${tonightData.temp2m}°.",
```
tonightData.temp2m is temperature of tonightTypicalWeather, not min/max temperature for the day.
You should add another property to compute min/max temperature of the day.

lngyeen commented on Flutter_Weather/lib/widgets/weather_detail_box_widget.dart:
```diff
@@ -0,0 +1,130 @@
+  Text todayText(){
+    WeatherData todayData = todayWeather.todayTypicalWeather;
+    return Text(
+      "Today is ${todayData.weather} with a high of ${todayData.temp2m}° "
```
tonightData.temp2m is temperature of todayTypicalWeather, not min/max temperature for the day.
You should add another property to compute min/max temperature of the day.

lngyeen commented on Flutter_Weather/lib/models/weather_extension.dart:
```diff
@@ -54,22 +136,105 @@ class DayWeather{
+    && initDate.add(Duration(hours: weather.timepoint)).hour > 15).toList();
+  }
+
+  WeatherData get tonightTypicalWeather{
```
tonightTypicalWeather should be the WeatherData with the most occurrences of the day.
(group `tonightWeathers` by `weather` property and pick the group with the largest number.)

lngyeen commented on Flutter_Weather/lib/models/weather_extension.dart:
```diff
@@ -54,22 +136,105 @@ class DayWeather{
+    return weathers.where((weather) => initDate.add(Duration(hours: weather.timepoint)).hour <15).toList(); 
+  }
+  
+  WeatherData get todayTypicalWeather{
```
todayTypicalWeather should be the WeatherData with the most occurrences of the day.
(group `todayWeathers ` by `weather` property and pick the group with the largest number.)

lngyeen commented on Flutter_Weather/lib/models/weather_extension.dart:
```diff
@@ -54,22 +136,105 @@ class DayWeather{
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(min);
+  
+  List<WeatherData> get todayWeathers{
+    return weathers.where((weather) => initDate.add(Duration(hours: weather.timepoint)).hour <15).toList(); 
```
I think time range of todayWeathers items should be from 6am -> 6pm. And time range of tonightWeathers items should be from 7pm -> 5am next day.

lngyeen commented on Flutter_Weather/lib/models/weather_extension.dart:
```diff
@@ -54,22 +136,105 @@ class DayWeather{
+    else return "to ${todayWeathers.first.windDirection}";
+  }
+
+  List<WeatherData> get tonightWeathers{
```
I think time range of todayWeathers items should be from 6am -> 6pm. And time range of tonightWeathers items should be from 7pm -> 5am next day.

heupotato committed:
Update weather_week_detail_widget.dart

heupotato committed:
Get weekData to pass to WeatherDetailBox Widget

heupotato committed:
Update weather_extension.dart

- Change time range of todayWeathers and tonightWeathers
- todayTypicalWeathers and tonightTypicalWeather get most occurrences of day/night
- Create getter to get maxTemp of daytime and minTemp of night

heupotato committed:
Update weather_detail_box_widget.dart

Change params due to previous changes
